### PR TITLE
Publish images in parallel

### PR DESCRIPTION
**Pull request by AndreasBurger**
**What this PR does / why we need it**:
Images will now be published (i.e. built and published) in parallel.

suggested by @rfranzke

### CODE REVIEW

**Comment by rfranzke** [comment]
Looking forward to this change!

File: concourse/steps/publish.mako [hunk 1]
```diff
@@ -1,7 +1,10 @@
-<%def name="publish_step(job_step, job_variant)", filter="trim">
+<%def name="publish_step(job_step, job_variant, indent)", filter="indent_func(indent),trim">
```
Comment (ccwienk) [inline concourse/steps/publish.mako:1]: why is it now required to introduce the `indent` func?

File: concourse/steps/publish.mako [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
 <%
+from makoutil import indent_func
 publish_trait = job_variant.trait('publish')
 %>
+# noop
```
Comment (ccwienk) [inline concourse/steps/publish.mako:6]: why noop?

File: concourse/templates/default.mako [hunk 3]
```diff
@@ -385,7 +385,7 @@ else:
 % elif job_step.name == 'update_component_dependencies':
         ${update_component_deps_step(job_step=job_step, job_variant=job_variant, github_cfg_name=github.name(), indent=8)}
 % elif job_step.name == 'publish':
-    ${publish_step(job_step=job_step, job_variant=job_variant)}
+        ${publish_step(job_step=job_step, job_variant=job_variant, indent=0)}
```
Comment (ccwienk) [inline concourse/templates/default.mako:388]: why did you change the indentation here?

Comment (rfranzke) [reply concourse/templates/default.mako:388]: Probably for consistency? see line 386

Comment (rfranzke) [inline concourse/templates/default.mako:388]: I understand too less/nothing of it, but shouldn't this be
```suggestion
        ${publish_step(job_step=job_step, job_variant=job_variant, indent=8)}
```

Comment (ccwienk) [reply concourse/templates/default.mako:388]: maybe.. however: before this change, this seems to have worked w/o using the indent function at all. so I wonder, why it should now suddenly be required to use it? Oo

Comment (ccwienk) [reply concourse/templates/default.mako:388]: good point

File: concourse/steps/publish.mako (hunk 2 above)
Comment (AndreasBurger) [reply concourse/steps/publish.mako:5]: this _seems_ to have been put in place for the preceding `prepare`-step:

```yaml
…
          run:
            path: /bin/sh
            args:
            - -exc
            - |
              # noop
      - in_parallel:
      
        - put: …
…
```

In any case, I've just preserved it.

File: concourse/templates/default.mako (hunk 3 above)
Comment (AndreasBurger) [reply concourse/templates/default.mako]: There are some strange things going on in this template. That 0-level indent effectively starts at level 6 (check the position of the `# noop` in my previous comment. It should be at level 0).

We really should refactor this template sometime. I _think_ it's in our backlog somewhere.

Comment (rfranzke) [reply concourse/templates/default.mako]: > We really should refactor this template sometime. I think it's in our backlog somewhere.

LOL :)

File: concourse/templates/default.mako [hunk 4]
```diff
@@ -385,7 +385,7 @@ else:
 % elif job_step.name == 'update_component_dependencies':
         ${update_component_deps_step(job_step=job_step, job_variant=job_variant, github_cfg_name=github.name(), indent=8)}
 % elif job_step.name == 'publish':
-    ${publish_step(job_step=job_step, job_variant=job_variant)}
+        ${publish_step(job_step=job_step, job_variant=job_variant)}
```
Comment (ccwienk) [inline concourse/templates/default.mako:388]: as discussed, this indent-change may be reverted
